src/lib.rs

```diff
@@ -840,4 +899,69 @@ mod test {
             err, AsyncHttpRangeReaderError::HttpError(err) if err.status() == Some(StatusCode::NOT_FOUND)
         );
     }
+
+    #[tokio::test]
+    async fn test_builder_happy_path() {
+        let path = Path::new(&std::env::var("CARGO_MANIFEST_DIR").unwrap()).join("test-data");
+        let server = StaticDirectoryServer::new(&path)
+            .await
+            .expect("could not initialize server");
+
+        let url = server.url().join("andes-1.8.3-pyhd8ed1ab_0.conda").unwrap();
+
+        // First, let's try to build the reader from a head response
+        let head_response = AsyncHttpRangeReader::initial_head_request(
+            Client::new(),
+            url.clone(),
+            HeaderMap::default(),
+        )
+        .await
+        .expect("could not perform head request");
+
+        let builder = AsyncHttpRangeReader::builder(Client::new().into())
+            .from_head_response(head_response)
+            .expect("could not build reader from head response")
+            .build()
+            .expect("could not build reader");
```

Comment:
I think this API is slightly weird. You first create an object followed by creating a builder. I would expect to be able to do both operations through the builder.